src/UnityPackage/Assets/Web3Unity/Scripts/Prefabs/ERC20/ERC20CustomTokenBalance.cs

```diff
@@ -10,7 +10,7 @@ async void Start()
         string contractAbi = "YOUR_TOKEN_ABI";
         // address of contract
         string contractAddress = "YOUR_TOKEN_ADDRESS";
-        var provider = new JsonRpcProvider("YOUR_NODE");
+        var provider = ProviderMigration.NewJsonRpcProvider("YOUR_NODE");
```

Comment:
We should be investigating if there's a way to avoid having this change. This one line impact every dev's project.

Reply:
True. I guess we could try to hack our way in with a new non-DI constructor, that utilizes static of some sort to populate his dependencies (like Web3Environment).
Still, parameter-type collisions are possible between the DI resolver and regular object construction with the new operator.
The first problem, I can think of, is editor suggestions for different JsonRpcProvider overloads. Some of them are there just to be used by DI, but users don't know about it.

Reply:
I could try implementing this, but this would require some time.

Reply:
It's possible to hide the DI-version of the constructor with [EditorBrowsable(EditorBrowsableState.Never)] attribute.
Still, for this feature to work in Rider I had to enable it in editor settings. Not sure if this is enabled by default, when you first install the IDE.
Should work in VS. Didn't test.

Reply:
Let's leave this for now, and we can get back to it once we start thinking about the compat layer.